source/utils/cache.js

```diff
@@ -11,15 +11,25 @@ const funcMap = {}; // Will there ever be a use case where this is a cache with
  * @return {Promise} - Promise either resolved with cached result of the function or rejected with function not found.
  */
 cache.resolveFunc = (key) => {
-  return cache.getAsync(key) // Can't do `cache.getAsync(key, true)` due to `get` argument ordering...
-    .catch({ errorcode: "ENOTFOUND" }, (e) => {
-      if (!funcMap[key]) throw e;     // func associated with key is not found, throw not found error
-      return getHardValue(funcMap[key].func()) // func is found, resolve, set with TTL and return result
-        .then((r) => {
-          return cache.setAsync(key, r, funcMap[key].ttl)
-            .then(() => { return r; })
-        });
-    });
+  return new Bluebird((resolve, reject) => {
+    let result = cache.get(key);
+    if (result !== undefined) {
+      resolve(result);
+      return;
+    }
+    result = funcMap[key];
+    if (result === undefined) {
+      reject(new Error(`Cache entry ${key} not found`));
+      return;
+    }
+    getHardValue(result.func()) // func is found, resolve, set with TTL and return result
+      .then((r) => {
+        cache.set(key, r, funcMap[key].ttl)
+        resolve(r);
+      }).catch((err) => {
+        reject(err);
+      });
+  });
```

Comment:
I'm not quite sure what this function does, but I think you can prevent the new-promise-antipattern like this:
```js
  let result = cache.get(key);
  if (result !== undefined) {
    return Bluebird.resolve(result);
  }
  result = funcMap[key];
  if (result === undefined) {
    return Bluebird.reject(new Error(`Cache entry ${key} not found`));
  }
  return getHardValue(result.func()) // func is found, resolve, set with TTL and return result
    .then((r) => {
      cache.set(key, r, funcMap[key].ttl)
      return r;
    });
```

Reply:
That's exactly what I had done first but then this test failed
https://github.com/FredrikNoren/ungit/blob/89a1d227935cb8772233e752a5e4c3be750e835a/test/spec.cache.js#L19
which throws an exception on `result.func()` which then isn't wrapped as a rejected promise...